### PR TITLE
Implemented support for transaction API.

### LINKED ISSUE #2: Support for transactions
```php
public function begin(): bool;
public function commit(): bool;
public function rollback(): bool;
```

### PR DESCRIPTION
**Pull request by freezemage0**
- Added `@throws` tags where needed.
- Removed unused static `PDODriver::$instance` property;
- Implemented `DriverException::createFromPdo()` named constructor for convenience.

Fixes #2 and #5

### CODE REVIEW

File: src/PDODriver.php [hunk 1]
```diff
@@ -62,4 +61,51 @@ public function isConnected(): bool
     {
         return isset($this->pdo);
     }
+
+    /**
+     * @throws DriverException
+     */
+    public function begin(): bool
```
Comment (xHeaven) [inline src/PDODriver.php:68]: Where is the `return`?

File: src/PDODriver.php [hunk 2]
```diff
@@ -62,4 +61,51 @@ public function isConnected(): bool
...
+            }
+        } catch (PDOException $e) {
+            throw DriverException::createFromPdo($e);
+        }
+    }
+
+    /**
+     * @throws DriverException
+     */
+    public function commit(): bool
```
Comment (xHeaven) [inline src/PDODriver.php:82]: Where is the `return`?

File: src/PDODriver.php [hunk 3]
```diff
@@ -62,4 +61,51 @@ public function isConnected(): bool
...
+            }
+        } catch (PDOException $e) {
+            throw DriverException::createFromPdo($e);
+        }
+    }
+
+    /**
+     * @throws DriverException
+     */
+    public function rollback(): bool
```
Comment (xHeaven) [inline src/PDODriver.php:96]: Where is the `return`?

**Review by xHeaven — CHANGES_REQUESTED** [review]

**Review by xHeaven — APPROVED** [review]
LGTM.

**Review by CommandString — APPROVED** [review]

**Pull request merged**